src/target-agents/base.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApiHelper } from '../helpers/api';
+
+export class TargetAgent extends ApiHelper {
+  public static friendlyName: string = '';
```

Comment:
You could make this class `abstract` to ensure at compile-time it's always extended rather than throwing run-time errors.

Reply:
This breaks L81 (`new this()`)